src/gpu/vulkan/SDL_gpu_vulkan.c

```diff
@@ -8186,6 +8186,124 @@ static void VULKAN_UploadToBuffer(
     VULKAN_INTERNAL_TrackBuffer(renderer, vulkanCommandBuffer, vulkanBuffer);
 }
 
+/* Readback */
+
+static void VULKAN_DownloadFromTexture(
+    SDL_GpuCommandBuffer *commandBuffer,
+    SDL_GpuTextureRegion *textureRegion,
+    SDL_GpuTransferBuffer *transferBuffer,
+    SDL_GpuBufferImageCopy *copyParams
+) {
+    VulkanCommandBuffer *vulkanCommandBuffer = (VulkanCommandBuffer*) commandBuffer;
+    VulkanRenderer *renderer = vulkanCommandBuffer->renderer;
+    VulkanTextureSlice *vulkanTextureSlice;
+    VulkanBufferContainer *transferBufferContainer = (VulkanBufferContainer*) transferBuffer;
+    VkBufferImageCopy imageCopy;
+    vulkanTextureSlice = VULKAN_INTERNAL_SDLToVulkanTextureSlice(&textureRegion->textureSlice);
+    VulkanResourceAccessInfo resourceAccessInfo;
+
+    resourceAccessInfo.stageMask = VK_PIPELINE_STAGE_TRANSFER_BIT;
+    resourceAccessInfo.accessMask = VK_ACCESS_TRANSFER_WRITE_BIT;
+    resourceAccessInfo.imageLayout = VK_IMAGE_LAYOUT_UNDEFINED;
```

Comment:
I'm surprised this doesn't have to be LAYOUT_TRANSFER_DST_OPTIMAL or something like that

Reply:
This is the buffer memory barrier, so the image layout doesn't apply. I'm considering splitting up the types for buffer access info and image access info to avoid this confusion